### DISCUSSION TITLE
qute-bitwarden userscript fails: "request_key: Required key not available"

### DISCUSSION THREAD

**Discussion by murchu27** [qutebrowser/qutebrowser#5940] [Q&A] [answered]
**Version info**:
qutebrowser v1.14.0
Git commit: 
Backend: QtWebEngine (Chromium 80.0.3987.163)
Qt: 5.15.0

Platform: Manjaro Linux

**Does the bug happen if you start with `--temp-basedir`?**:
Yes

**Description**
Trying to run the `qute-bitwarden` userscript with the command:

    spawn --userscript qute-bitwarden

but I'm getting the following error:

```
Process stderr:
request_key: Required key not available
Traceback (most recent call last):
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 302, in <module>
    sys.exit(main(arguments))
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 219, in main
    pass_(
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 149, in pass_
    session_key = get_session_key(auto_lock)
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 132, in get_session_key
    session = ask_password()
  File "/usr/share/qutebrowser/userscripts/qute-bitwarden", line 102, in ask_password
    process = subprocess.run([
  File "/usr/lib/python3.8/subprocess.py", line 489, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'rofi'
```

I've installed bitwarden cli according to [these instructions](https://bitwarden.com/help/article/cli/#npm), and logged in. I tried exporting my session key as an environment variable, and running qutebrowser from that terminal, but the error appears anyway.

What can I do here to get this working?

**Comment by toofar** [comment, marked as answer]
> FileNotFoundError: [Errno 2] No such file or directory: 'rofi'

It looks like it is just complaining about not being able to find rofi?
It looks like it caches the session with keyctl and that's probably what the "Required key not available" message is from. Then it falls back on using rofi to ask for your master password. So that message is expected on first run.
https://github.com/qutebrowser/qutebrowser/blob/master/misc/userscripts/qute-bitwarden#L26

**Comment by The-Compiler** [comment]
Moving this to a discussion since this isn't an issue in qutebrowser - I think @toofar's answer is spot on, though.

**Comment by murchu27** [comment]
Thanks for that folks. Getting another issue now, but it might not apply to others who see this post, so I'll open a new discussion for that.
